lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -282,97 +398,167 @@ def parse_genbank(self, file_path, params):
 
+    def _get_contigs_and_extra_info(self, contigs, genome_obj):
         out_contigs = []
         extra_info = defaultdict(dict)
         for in_contig in contigs:
             if in_contig.annotations.get('topology', "") == 'circular':
                 extra_info[in_contig.id]['is_circ'] = 1
-                self.circ_contigs.add(in_contig.id)
+                genome_obj.circ_contigs.add(in_contig.id)
             elif in_contig.annotations.get('topology', "") == 'linear':
                 extra_info[in_contig.id]['is_circ'] = 0
             out_contigs.append(in_contig)
-            self.contig_seq[in_contig.id] = in_contig.seq.upper()
-
-        assembly_ref = params.get("use_existing_assembly")
-        if assembly_ref:
-            if not re.match("\d+\/\d+\/\d+", assembly_ref):
-                raise ValueError(f"Assembly ref: {assembly_ref} is not a valid format. Must"
-                                 f" be in numerical <ws>/<object>/<version> format.")
-            ret = self.dfu.get_objects(
-                {'object_refs': [assembly_ref]}
-            )['data'][0]
-            if "KBaseGenomeAnnotations.Assembly" not in ret['info'][2]:
-                raise ValueError(f"{assembly_ref} is not a reference to an assembly")
-            unmatched_ids = list()
-            unmatched_ids_md5s = list()
-            for current_contig in self.contig_seq.keys():
-                current_contig_md5 = hashlib.md5(
-                    str(self.contig_seq[current_contig]).encode('utf8')
-                ).hexdigest()
-                if current_contig in ret['data']['contigs']:
-                    if current_contig_md5 != ret['data']['contigs'][current_contig]['md5']:
-                        unmatched_ids_md5s.append(current_contig)
-                else:
-                    unmatched_ids.append(current_contig)
-            if len(unmatched_ids) > 0:
-                raise ValueError(warnings['assembly_ref_extra_contigs'].format(", ".join(unmatched_ids)))
-            if len(unmatched_ids_md5s) > 0:
-                raise ValueError(warnings["assembly_ref_diff_seq"].format(", ".join(unmatched_ids_md5s)))
-            logging.info(f"Using supplied assembly: {assembly_ref}")
-            return assembly_ref
-        logging.info("Saving sequence as Assembly object")
-        Bio.SeqIO.write(out_contigs, fasta_file, "fasta")
-        assembly_ref = self.aUtil.save_assembly_from_fasta(
-            {'file': {'path': fasta_file},
-             'workspace_name': params['workspace_name'],
-             'assembly_name': assembly_id,
-             'type': params.get('genome_type', 'isolate'),
-             'contig_info': extra_info})
-        logging.info(f"Assembly saved to {assembly_ref}")
-        return assembly_ref
+            genome_obj.contig_seq[in_contig.id] = in_contig.seq.upper()
+        return out_contigs, extra_info
+
+    def _get_contigs_and_validate_existing_assembly(self, genome_objs):
```

Comment:
```suggestion
    def _load_contigs_for_and_validate_existing_assemblies(self, genome_objs):
```

I was really confused when this method only worked on genomes where the assembly already existed

Reply:
👍

Reply:
:+1: